apps/server/src/modules/tool/controller/tool.controller.spec.ts

```diff
@@ -319,4 +334,103 @@ describe('ToolController', () => {
 			});
 		});
 	});
+
+	describe('findExternalTool', () => {
+		function setupFind() {
+			const filterQuery: ExternalToolSearchParams = new ExternalToolSearchParams();
+			const pagination: PaginationParams = { skip: 0, limit: 1 };
+			const sortingQuery: SortExternalToolParams = { sortOrder: SortOrder.asc, sortBy: ExternalToolSortOrder.NAME };
+			const filter: IFindOptions<ExternalToolDO> = {
+				pagination: { skip: 0, limit: 1 },
+				order: { name: SortOrder.asc },
+			};
+			const currentUser: ICurrentUser = { userId: 'userId' } as ICurrentUser;
+			const externalToolDO: ExternalToolDO = new ExternalToolDO({
+				id: '1',
+				name: 'mockName',
+				url: 'mockUrl',
+				logoUrl: 'mockLogoUrl',
+				parameters: [],
+				isHidden: true,
+				openNewTab: true,
+				version: 1,
+				config: new Oauth2ToolConfigDO({} as Oauth2ToolConfigDO),
```

Comment:
is this cast neccessary? you could use basic, if you want to write less, instead.